zipline-loader/src/androidMain/kotlin/app/cash/zipline/loader/internal/database/databaseAndroid.kt

```diff
@@ -13,20 +13,26 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
-package app.cash.zipline.loader
+package app.cash.zipline.loader.internal.database
 
+import android.content.Context
+import android.database.SQLException
 import com.squareup.sqldelight.android.AndroidSqliteDriver
 import com.squareup.sqldelight.db.SqlDriver
+import okio.Path
 
-actual class DriverFactory(private val context: android.content.Context) {
-  actual fun createDriver(): SqlDriver {
+internal actual class SqlDriverFactory(
+  private val context: Context,
+) {
+  actual fun create(path: Path, schema: SqlDriver.Schema): SqlDriver {
+    validateDbPath(path)
     return AndroidSqliteDriver(
-      schema = Database.Schema,
+      schema = schema,
       context = context,
-      name = "zipline-loader.db",
+      name = path.toString(),
```

Comment:
Ooooh good catch. I changed `useNoBackupDirectory` to `false` so that it would honor the full path. Otherwise the path shouldn’t be fully-qualified.
https://android.googlesource.com/platform/frameworks/support/+/refs/heads/androidx-main/sqlite/sqlite-framework/src/main/java/androidx/sqlite/db/framework/FrameworkSQLiteOpenHelper.java?source=post_page---------------------------%2F%2F%2F%2F%2F&autodive=0%2F#93